tests/news/tests.py

```diff
@@ -0,0 +1,155 @@
+from contextlib import contextmanager
+from pathlib import Path
+
+from django.contrib.admin.tests import AdminSeleniumTestCase
+from django.contrib.auth.models import User
+from django.test import modify_settings, override_settings
+from django.test.selenium import ChangeWindowSize, screenshot_cases
+from django.urls import reverse
+from django.utils.text import slugify
+
+from .models import Article
+
+SIZES = ["docs_size"]
+
+
+@modify_settings(INSTALLED_APPS={"append": "django.contrib.flatpages"})
+@override_settings(ROOT_URLCONF="news.urls")
+class SeleniumTests(AdminSeleniumTestCase):
+    available_apps = AdminSeleniumTestCase.available_apps + [
+        "django.contrib.flatpages",
+        "news",
+    ]
+
+    def setUp(self):
+        self.superuser = User.objects.create_superuser(
+            username="admin",
+            password="secret",
+            email="admin@example.com",
+            first_name="Admin",
+            last_name="Super",
+        )
+        self.admin_login(username="admin", password="secret")
+
+    @contextmanager
+    def docs_size(self):
+        with ChangeWindowSize(800, 800, self.selenium):
+            yield
+
+    def take_screenshot(self, name):
+        path = Path.cwd().parent / "docs" / "ref" / "contrib" / "admin" / "_images"
+        assert path.exists()
+        self.selenium.save_screenshot(path / f"{name}.png")
+
+    def hide_nav_sidebar(self):
+        """Hide the navigation sidebar if it's open."""
+        from selenium.webdriver.common.by import By
+
+        if self.selenium.find_element(By.ID, "nav-sidebar").is_displayed():
+            self.selenium.find_element(By.CSS_SELECTOR, "#toggle-nav-sidebar").click()
+        self.selenium.execute_script(
+            'document.getElementById("toggle-nav-sidebar").style.display = "none";'
+        )
+
+    def change_header_display(self, display):
+        self.selenium.execute_script(
+            f'document.getElementById("header").style.display = "{display}";'
+        )
+        self.selenium.execute_script(
+            f'document.getElementsByTagName("nav")[0].style.display = "{display}";'
+        )
+
+    @screenshot_cases(SIZES)
```

Comment:
As `take_screenshot` has been overwritten to always take a screenshot, I think we should use the `ChangeWindowSize` context manager directly. 
I suspect that because we change the window size happens first and then the top bar is hidden after, the screenshots are smaller than expected. I might be wrong, but if we hide first and then resize the screen, we might get closer to 800x800px
Also in some cases (such as the flat pages one) we may want to increase the screenshot size slightly, so this might give us more flexibility

EDIT: I see you've already mentioned this in your PR summary 👍